Forms/MainForm.cs

```diff
@@ -75,12 +78,22 @@ private void UpdateConfigurationSubstitutions()
         {
             Configuration.DefaultSubstitutions.Clear();
             Configuration.DefaultSubstitutions.AddRange(substitutionList1.Substitutions);
+            LoadConfigurationDetails();
         }
 
 
         private void LoadConfigurationDetails()
         {
-            txtWordList.Text = Configuration.DefaultWords.ToString();
+            string wordtext = Configuration.DefaultWords.ToString();
+            if (txtWordList.Text != null && txtWordList.Text != wordtext && txtWordList.Text != wordtext + " ")

```

Comment:
I still had an issue with the PR version where when I would add a second space, the field would reload and reposition the cursor...

You are right, basing the operation only on the focus might skip the initial load, we can probably add a check with the input's text length... 

If you feel an other approach might not be as straight forward as I expect, I can approve this PR and work on this later on... At least for now, the UI experience is a lot better than what it was before.